--- Acse.y	2017-12-12 12:04:19.000000000 +0100
+++ Original.y	2018-12-18 08:47:16.727918992 +0100
@@ -124,6 +124,7 @@
 %token RETURN
 %token READ
 %token WRITE
+%token MERGE
 
 %token <label> DO
 %token <while_stmt> WHILE
@@ -570,6 +571,27 @@
                                  (program, exp_r0, $2, SUB);
                         }
                      }
+   |MERGE exp COMMA exp COMMA exp {
+	if($6.expression_type==IMMEDIATE) {
+		if($6.value>0) $$=$4;
+		else $$=$2;
+	}
+	else{
+	   int location;
+	   t_axe_label *label1,*label2;
+	   int reg=getNewRegister(program);
+	   gen_add_instruction(program,reg,REG_0,$6.value,CG_DIRECT_ALL);
+	   label1=newLabel(program);
+	   label2=newLabel(program);
+	   gen_beq_instruction(program,label1,0);
+	   gen_add_instruction(program,reg,REG_0,$4.value,CG_DIRECT_ALL);		/* DIFFERENT FROM ZERO WE ASSIGN THE VALUE OF THE SECOND EXPRESSION */
+	   gen_bt_instruction(program,label2,0);
+	   assignLabel(program,label1);
+	   gen_add_instruction(program,reg,REG_0,$2.value,CG_DIRECT_ALL);		/* EQUAL TO ZERO we assign the first value */
+	   assignLabel(program,label2);
+	   $$=create_expression(reg,REGISTER);
+	  }
+	}
 ;
 
 %%
