--- Acse.y	2019-01-12 15:01:46.919644011 +0100
+++ Original.y	2017-12-12 12:04:19.000000000 +0100
@@ -108,7 +108,6 @@
    t_list *list;
    t_axe_label *label;
    t_while_statement while_stmt;
-   t_questionmark questionamrk;
 } 
 /*=========================================================================
                                TOKENS 
@@ -133,10 +132,6 @@
 %token <intval> TYPE
 %token <svalue> IDENTIFIER
 %token <intval> NUMBER
-%token <questionamrk> QUESTION
-
-
-/* DA ERRORE MA é UGUALE ALLA SOLUZOONE   */
 
 %type <expr> exp
 %type <decl> declaration
@@ -158,7 +153,6 @@
 %left SHL_OP SHR_OP
 %left MINUS PLUS
 %left MUL_OP DIV_OP
-%right QUESTION
 %right NOT
 
 /*=========================================================================
@@ -477,29 +471,6 @@
                      /* free the memory associated with the IDENTIFIER */
                      free($1);
    }
-	 
-   | exp QUESTION {
-	$2.end=newLabel(program);
-	  $2.second=newLabel(program);
-	
-	  if ($1.expression_type == IMMEDIATE)
-                         gen_load_immediate(program, $1.value);
-                     else
-                         gen_andb_instruction(program, $1.value,
-                             $1.value, $1.value, CG_DIRECT_ALL);
-
-	  gen_beq_instruction(program,$2.second,0);
-	}
-	exp {
-	  $$=$4;
-	  gen_bt_instruction(program,$2.end,0);
-	  assignLabel(program,$2.second);
-	}
-	COLON exp{
-	  $$=$7;
-	  assignLabel(program,$2.end);
-	}
-
    | IDENTIFIER LSQUARE exp RSQUARE {
                      int reg;
                      
